pkg/ocp/machinesets.go

```diff
@@ -132,3 +136,27 @@ func (msd *k8sMachineSetDeployer) Delete(machineSet *unstructured.Unstructured)
 
 	return errors.Wrapf(err, "error deleting machine set %q", machineSet.GetName())
 }
+
+func (msd *k8sMachineSetDeployer) List(machineSet *unstructured.Unstructured, name string) ([]unstructured.Unstructured, error) {
```

Comment:
Occasionally we are hearing issues where the length of the names is creating problems as it exceeds the max permissible value. And also in some clouds we use the prefix, "submariner-gw-" and i remember using some shorter prefix in other clouds. It might be better to search for labels on the machineSet instead of relying on the names. Basically it will help in future during upgrades if we have to modify the submariner-gateway node names due to some length constraints. Your thoughts?

Reply:
We don't have machineset, but we have it inside the spec in the machineset, Do you mean to use that label instead of name?

Added the name as a parameter since the caller can use it as per the naming convention used for the cloud.

Reply:
>We don't have machineset, but we have it inside the spec in the machineset, Do you mean to use that label instead of name?

Yes. Using the label will be agnostic to the changes we make in the naming convention. But if you think we are handing it properly even with names, I'm fine.

Reply:
Yes, the name is intended to do the same.